extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -312,10 +362,12 @@ public void run()
               public ConsumerRecords<byte[], byte[]> call() throws Exception
               {
                 try {
+                  status = Status.READING;
                   return consumer.poll(POLL_TIMEOUT);
                 }
-                finally {
-                  startedReading = true;
+                catch (OffsetOutOfRangeException e) {
+                  status = Status.WAITING;
```

Comment:
what does the `READING -> WAITING -> READING -> WAITING -> …` loop give us over just having `READING`?